### PR TITLE
Add dynamic mode for scripted avatar motor

### PR DESCRIPTION
**Pull request by dback2**
Add a motorMode property which is "simple" by default which behaves as it does now using the motorTimescale property for motor's timescale, or alternatively "dynamic" mode which uses the action's horizontal and vertical timescales instead allowing for stepping up and braking.

Test Plan:
https://highfidelity.testrail.net/index.php?/cases/view/11887

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7454/app_12485_PR.apk

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -1778,6 +1816,14 @@ void MyAvatar::setScriptedMotorFrame(QString frame) {
     }
 }
 
+void MyAvatar::setScriptedMotorMode(QString frame) {
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyAvatar.cpp]: Change `frame` variable name to be `mode` for better code readability.  Here and also in the declaration.

File: interface/src/avatar/MyAvatar.h [hunk 2]
```diff
@@ -69,6 +69,7 @@ class MyAvatar : public Avatar {
      * @property motorTimescale {float} Specifies how quickly the avatar should accelerate to meet the motorVelocity,
      *   smaller values will result in higher acceleration.
      * @property motorReferenceFrame {string} Reference frame of the motorVelocity, must be one of the following: "avatar", "camera", "world"
+     * @property motorMode {string} Type of scripted motor behavior, "simple" = unmodified legacy behavior and "dynamic" = same as default motor
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyAvatar.h]: "simple" isn't really "legacy" so much as... it doesn't dynamically adjust the motor timescales according to step-up logic and hover state.  Instead its timescale is whatever the script sets (see?  it is "simple").  Dunno how best to state this, but the doc comment here needs work.

The doc comment should also indicate that "simple" is the default option.  Not to be confused with the non-scripted "default" motor.  Maybe you shouldn't call it the "default" motor?  It needs a better name.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7454/HighFidelity-Beta-PR12485-ba0ba96d81a096100948d11730b3ff2d7b3825a3.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7454/HighFidelity-Beta-PR12485-ba0ba96d81a096100948d11730b3ff2d7b3825a3.dmg

File: interface/src/avatar/MyAvatar.cpp (hunk 1 above)
Comment (dback2) [reply interface/src/avatar/MyAvatar.cpp]: Woops you're right missed that

File: interface/src/avatar/MyAvatar.h (hunk 2 above)
Comment (dback2) [reply interface/src/avatar/MyAvatar.h]: True "legacy" is probably not the best use of words here. "Unmodified" might still make sense but maybe there is something better for that fact that it is not adjusting timescales or doing step ups? Would "Unmodified behavior and default option" be good?

In terms of the default motor, I see elsewhere we refer to action motor vs scripted motor so perhaps could stick to that naming for consistency.

File: interface/src/avatar/MyAvatar.cpp [hunk 3]
```diff
@@ -1623,32 +1626,38 @@ controller::Pose MyAvatar::getControllerPoseInAvatarFrame(controller::Action act
...
+    float horizontalMotorTimescale;
+    float verticalMotorTimescale;
 
-        float horizontalMotorTimescale;
-        float verticalMotorTimescale;
+    if (_characterController.getState() == CharacterController::State::Hover ||
+            _characterController.computeCollisionGroup() == BULLET_COLLISION_GROUP_COLLISIONLESS) {
+        horizontalMotorTimescale = FLYING_MOTOR_TIMESCALE;
+        verticalMotorTimescale = FLYING_MOTOR_TIMESCALE;
+    }
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.cpp]: Brace placement doesn't match coding standard.

File: interface/src/avatar/MyAvatar.cpp [hunk 4]
```diff
@@ -1660,15 +1669,36 @@ void MyAvatar::updateMotors() {
...
-            motorRotation = getMyHead()->getHeadOrientation() * glm::angleAxis(PI, Vectors::UNIT_Y);
-        } else if (_scriptedMotorFrame == SCRIPTED_MOTOR_AVATAR_FRAME) {
-            motorRotation = getWorldOrientation() * glm::angleAxis(PI, Vectors::UNIT_Y);
-        } else {
-            // world-frame
-            motorRotation = glm::quat();
+        if (_scriptedMotorMode == SCRIPTED_MOTOR_SIMPLE_MODE) {
+            if (_scriptedMotorFrame == SCRIPTED_MOTOR_CAMERA_FRAME) {
+                motorRotation = getMyHead()->getHeadOrientation() * glm::angleAxis(PI, Vectors::UNIT_Y);
+            }
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.cpp]: Brace placement doesn't match coding standard.

File: interface/src/avatar/MyAvatar.cpp [hunk 5]
```diff
@@ -1660,15 +1669,36 @@ void MyAvatar::updateMotors() {
...
+            else if (_scriptedMotorFrame == SCRIPTED_MOTOR_AVATAR_FRAME) {
+                motorRotation = avatarOrientation * glm::angleAxis(PI, Vectors::UNIT_Y);
+            }
+            else {
+                // world-frame
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.cpp]: Brace placement doesn't match coding standard.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7492/app_12485_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7493/app_12485_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7494/app_12485_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7494/HighFidelity-Beta-PR12485-283b5724ba42b69edec649ed15a0a60335c29e1d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7494/HighFidelity-Beta-PR12485-283b5724ba42b69edec649ed15a0a60335c29e1d.exe

**Comment by conklin94122** [comment]
Removing QA Ready until changes are in place. Please reapply the label when it's ready for testing.

**Comment by qlimnavarro** [comment]
Currently Testing.

**Comment by qlimnavarro** [comment]
Pass. @conklin94122

**Pull request merged**